### PR TITLE
Regression in asObject

### PR DESCRIPTION
**Pull request by pontusmelke**
Completely relying on the value mapper introduced a regression for some queries

### CODE REVIEW

**Review by henriknyman — APPROVED** [review]

File: community/kernel/src/main/java/org/neo4j/kernel/impl/util/DefaultValueMapper.java [hunk 1]
```diff
@@ -75,22 +80,172 @@ public Path mapPath( PathValue value )
...
+                public boolean hasNext()
+                {
+                    return index < size;
+                }
+
+                @Override
+                public PropertyContainer next()
+                {
+                    PropertyContainer propertyContainer;
+                    if ( index % 2 == 0 )
```
Comment (henriknyman) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/util/DefaultValueMapper.java]: Not important, but could be `index & 1`

File: community/kernel/src/main/java/org/neo4j/kernel/impl/util/DefaultValueMapper.java [hunk 2]
```diff
@@ -75,22 +80,172 @@ public Path mapPath( PathValue value )
...
+                    {
+                        propertyContainer = mapNode( nodes[index / 2] );
```
Comment (henriknyman) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/util/DefaultValueMapper.java]: Not important, but could be `index >> 1`

File: community/kernel/src/main/java/org/neo4j/kernel/impl/util/DefaultValueMapper.java [hunk 3]
```diff
@@ -75,22 +80,172 @@ public Path mapPath( PathValue value )
...
+                    }
+                    else
+                    {
+                        propertyContainer = mapRelationship( relationships[index / 2] );
```
Comment (henriknyman) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/util/DefaultValueMapper.java]: Not important, but could be `index >> 1`

**Review by henriknyman — APPROVED** [review]

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ToStringAcceptanceTest.scala [hunk 4]
```diff
@@ -58,14 +58,6 @@ class ToStringAcceptanceTest extends ExecutionEngineFunSuite with Matchers {
     }
   }
 
-  test("Path should provide sensible toString") {
```
Comment (pontusmelke) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/ToStringAcceptanceTest.scala]: Let's talk about this. Should we really expect this to work?

**Pull request merged**